tests/phpunit/integration/Core/Util/User_Input_SettingsTest.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Core\Util\User_Input_SettingsTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Util;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Core\Util\User_Input_Settings;
+use Google\Site_Kit\Tests\Core\Util\FakeUser_Input_Settings;
+use Google\Site_Kit\Tests\TestCase;
+
+class User_Input_SettingsTest extends TestCase {
+
+	/**
+	 * Context object.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * @var bool
+	 */
+	protected static $old_wp_using_ext_object_cache;
+
+	public static function wpSetUpBeforeClass() {
+		self::$old_wp_using_ext_object_cache = wp_using_ext_object_cache();
+	}
+
+	public static function wpTearDownAfterClass() {
+		wp_using_ext_object_cache( self::$old_wp_using_ext_object_cache );
+	}
```

Comment:
I think the external cache configuration changes shouldn't be necessary here unless another test isn't cleaning up after itself properly. Maybe we could simply add an assertion in the test that `assertFalse( wp_using_ext_object_cache() )` ?

If you would prefer to keep it, it's okay with me, just a thought.

Reply:
I would prefer to keep it and make sure that we don't use an external cache.